apps/spruce/src/pages/version/Tabs/index.tsx

```diff
@@ -184,67 +182,63 @@ export const VersionTabs: React.FC<VersionTabProps> = ({ version }) => {
       numFailedChildPatches,
       numStartedChildPatches,
       numSuccessChildPatches,
-      versionId: versionId ?? "",
+      versionId: version.id,
     });
-  }, [taskCount, childPatches, versionId]);
+  }, [taskCount, childPatches, version.id]);
 
   const activeTabs = useMemo(
-    // @ts-expect-error: FIXME. This comment was added by an automated script.
-    () => Object.keys(allTabs).filter((t) => tabIsActive[t] as PatchTab[]),
+    () =>
+      (Object.keys(allTabs) as VersionPageTabs[]).filter((t) => tabIsActive[t]),
     [allTabs, tabIsActive],
   );
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  const isValidTab = tabIsActive[tab];
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  const [selectedTab, setSelectedTab] = useState(activeTabs.indexOf(tab));
-  const previousTab = usePrevious(selectedTab);
+  const [selectedTab, setSelectedTab] = useState(tab || VersionPageTabs.Tasks);
 
-  useEffect(() => {
-    // If tab is not valid, set to task tab.
-    if (!isValidTab) {
-      // @ts-expect-error: FIXME. This comment was added by an automated script.
-      navigate(getVersionRoute(versionId), { replace: true });
+  const handleTabChange = (
+    newTab: VersionPageTabs,
+    sendAnalytics: boolean = false,
```

Comment:
I think it makes sense to make it a required parameter since the argument is always being provided
```suggestion
    sendAnalytics: boolean,
```